### ISSUE TITLE
chainer v1.11.0 ptb example training with --test option

### ISSUE THREAD

**Issue by odanado** [chainer/chainer#1384]
I run ptb example with --test option:

```
./train_ptb.py -g 0 --test
```

Screen during training:

```
#vocab = 10000
     total [##############################################################] 125.64%
this epoch [..................................................]  0.00%
         7 iter, 49 epoch / 39 epochs
   0.15797 iters/sec. Estimated time to finish: -1 day, 23:59:50.956683.
```

result is:

```
#vocab = 10000
epoch       iteration   perplexity  val_perplexity
70          10          3918.07     43725.3
     total [###############################################################################################################################################] 287.18     total [#######################################################################################################################################################     total [#######################################################################################################################################################     total [#######################################################################################################################################################140         20          1003.55     45304.3
     total [#######################################################################################################################################################     total [#######################################################################################################################################################     total [#######################################################################################################################################################     total [#######################################################################################################################################################     total [#######################################################################################################################################################     total [#######################################################################################################################################################     total [#######################################################################################################################################################     total [#######################################################################################################################################################     total [#######################################################################################################################################################     total [#######################################################################################################################################################210         30          205.992     13718.8
     total [#######################################################################################################################################################     total [#######################################################################################################################################################     total [#######################################################################################################################################################     total [#######################################################################################################################################################     total [#######################################################################################################################################################     total [##########################################################################################################################################################################################################################################################################################################################     total [##########################################################################################################################################################################################################################################################################################################################     total [##########################################################################################################################################################################################################################################################################################################################     total [##########################################################################################################################################################################################################################################################################################################################     total [##########################################################################################################################################################################################################################################################################################################################test
test perplexity: 28380.350504
```

But when the number of epochs is multiple of 7 (e.g. 35), training is a success:

```
./train_ptb.py -g 0 --test -e 35
```

result: 

```
#vocab = 10000
test
test perplexity: 4670.25469318
```

**Comment by Smerity** [comment]
This is also a problem when it comes to the proper training as well, not just `--test`, so it's likely an issue with the epoch iterator or similar?

**Comment by kissge** [comment]
Until this bug would be fixed, how about changing the default value to 35, not 39?
Otherwise new users might get confused.

**Comment by delta2323** [comment]
Now that #1423 is merged, the script stops even if the number of epochs is not divisible by 7.

On the other hand, current implementation occurs another problem. As the trainer consumes 42 epochs in total, which is larger than prescribed total epoch, some numbers of `ProgressBar` goes wrong (e.g. the percentile exceeds 100% or estimated time is negative)

```
%(master)[]./train_ptb.py --test
#vocab = 10000
     total [#####################################################] 107.69%
this epoch [..................................................]  0.00%
         6 iter, 42 epoch / 39 epochs
  0.068392 iters/sec. Estimated time to finish: -1 day, 23:59:53.733622.
```

The problem derives from the current implementation of `Iterator` , as I [commented](https://github.com/pfnet/chainer/pull/1423#issuecomment-251549850) in #1423.
There are two options to fix this problem:
(1) Fix dataset iterators so that it does not retrieve exceeding dataset (or add option that switches the behavior to the iterators).
(2) Fix `ProgressBar` so that it accepts the situation in which current epoch number can exceed the number of total epochs.

I am not certain we should limit the last minibatch so that it does not exceeds the total epoch or

**Comment by delta2323** [comment]
Solved via #1384, #2347, 2484

**Comment by delta2323** [comment]
Although the 42/39 problem remains, this topic different from the one addressed in this issue. I made a new issue about the problem and close this issue. Thank you.
